File: Modul-6/soal-4/PRAK604-2410817210017-OrlandoSugian.c
#include <stdio.h>
#include <string.h>
int main() {
    char kode[100], pesan[100];
    int bintang = 0, pagar = 0;
    fgets(kode, sizeof(kode), stdin);
    kode[strcspn(kode, "\n")] = '\0';
    fgets(pesan, sizeof(pesan), stdin);
    pesan[strcspn(pesan, "\n")] = '\0'; 
    if (strlen(kode) != strlen(pesan)) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    }
    printf("Hasil: ");
    for (int i = 0; i < strlen(kode); i++) {
        if (kode[i] == pesan[i]) {
            printf("*");
            bintang++;
        } else {
            printf("#");
            pagar++;
        }
    }
    printf("\n");
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);
    if (bintang >= pagar) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }
    return 0;
}

File: Modul-6/soal-4/PRAK604-2410817210017-OrlandoSugian.py
# Memasukkan kode dan pesan
kode = input("")
pesan = input("")
if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
else:
    bintang = 0
    pagar = 0
    hasil = []
    for k, p in zip(kode, pesan):
        if k == p:
            hasil.append("*")
            bintang += 1
        else:
            hasil.append("#")
            pagar += 1
    print("".join(hasil))
    print(f"* = {bintang}")
    print(f"# = {pagar}")
    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")
